xbmc/pvr/PVRManager.cpp

```diff
@@ -113,23 +113,38 @@ CPVRManager::~CPVRManager(void)
 
 void CPVRManager::Announce(AnnouncementFlag flag, const char *sender, const char *message, const CVariant &data)
 {
-  if (!IsStarted() || (flag & (System)) == 0)
+  if (!IsStarted())
     return;
 
-  if (strcmp(message, "OnWake") == 0)
+  if ((flag & (ANNOUNCEMENT::System)))
+  {
+    if (strcmp(message, "OnSleep") == 0)
```

Comment:
@janbar Yeah, unfortunately, changing it the way you did would have exactly the problem we are trying to solve with removal if the old api function.